Mathlib/CategoryTheory/Category/GaloisConnection.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2017 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stephen Morgan, Scott Morrison, Johannes Hölzl, Reid Barton
+
+! This file was ported from Lean 3 source module category_theory.category.galois_connection
+! leanprover-community/mathlib commit d82b87871d9a274884dff5263fa4f5d93bcce1d6
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Category.Preorder
+import Mathlib.CategoryTheory.Adjunction.Basic
+import Mathlib.Order.GaloisConnection
+
+/-!
+
+# Galois connections between preorders are adjunctions.
+
+* `GaloisConnection.adjunction` is the adjunction associated to a galois connection.
+
+-/
+
+
+universe u v
+
+section
+
+variable {X : Type u} {Y : Type v} [Preorder X] [Preorder Y]
+
+/-- A galois connection between preorders induces an adjunction between the associated categories.
+-/
+def GaloisConnection.adjunction {l : X → Y} {u : Y → X} (gc : GaloisConnection l u) :
+    gc.monotone_l.functor ⊣ gc.monotone_u.functor :=
+  CategoryTheory.Adjunction.mkOfHomEquiv
+    { homEquiv := fun X Y =>
+        ⟨fun f => CategoryTheory.homOfLE (gc.le_u f.le),
```

Comment:
Here I had to change `.hom` to `homOfLE`, otherwise the linter complaint about def being `noncomputable`. It somehow seems like the linter detects alias of somthing computable as noncomputable? That'd be weird, but changing here to `homOfLE` seems like a harmless workaround